common/src/sql/dbinit.sql

```diff
@@ -489,6 +489,16 @@ CREATE UNIQUE INDEX on omicron.public.image (
 ) WHERE
     time_deleted is NULL;
 
+CREATE TYPE omicron.public.distribution AS ENUM (
+  'alpine',
+  'debian',
+  'ubuntu',
+  'rocky',
+  'centos',
+  'fedora',
+  'freebsd'
+);
```

Comment:
I have some hesitation about constraining a list of distributions in the database. I can see us having a table of distributions where we provide some defaults, but I don't think we generally want to be tasked with updating this everytime someone requests a new distribution.

Reply:
If someone's requesting a new Oxide supplied OS distribution to be packaged and distributed, I'd want that to be an explicit add. I'd rather that than a freeform string here.

If the customer wants to upload their own custom image, that is another deal entirely. Now that I'm typing this out, we're making a distinction between Oxide supplied global images and Customer created project images, but I think there's a fit for Customer created global images as well (a Debian base image that everyone should use, for example).

Reply:
Why do we need to care about the difference between a customer created image and one that Oxide has created?

From an end user perspective, our specific Oxide things won't be particularly important versus whatever the project has standardized on.

Reply:
Sorry, let me write this differently - I'm not saying that Nexus has to care, or that there should be a distinction between them, but that we should write our authz policy to 

1) allow customers to POST to `/images` to create their own global base images
2) not allow customers to delete our Oxide supplied global base images

Reply:
I don't get why (2) should be true. Oxide supplied global base images don't come with the rack at all. Maybe one will be there to speed up the provisioning process; however, these are just something that we should think of as being able to import into the environment the same way that operators did. If an operator imported say alpine version x and ubuntu version y and then import version x+1 and y+1, they should certainly be able to remove alpine x and unbutu y from their rack/az. Similarly if they mis-import something they should be able to delete it as well. It seems weird to say oh, this came from Oxide, this space for something you never want to use or you never want projects to use shouldn't be reclaimable. Can you describe a bit more about why someone shouldn't be able to delete images they import from Oxide from their environment?

Reply:
I agree with the arguments here about why (2) shouldn't be true.

I'm imaging the root of the distinction between Oxide supplied images and user images being that Oxide images will be validated in some way that user images will not be, through something like a signature. Oxide images may also come to the rack through a different method, like from the update system instead of being uploaded by the user for example. I'm thinking that a distinction has to exist though maybe not applicable to authz policy - users could delete Oxide supplied global images or their own.

Reply:
we really need Oxide supplied defaults. Operators will want to get a rack and have it usable to deploy VMs from day one. Just like other cloud providers provide base images for ubuntu, debian, common things, we too will do the same. The ownus should not be on the end user as that is a huge amount of time spent on their part to get images then before they can even deploy a VM.

Reply:
I'm not saying we shouldn't have defaults here that folks can pull and use. The question I'm asking is should they actually be treated differently and be special from a database perspective.

Reply:
Yeah I don't have an opinion on that :)